src/streamlink/plugins/vlive.py

```diff
@@ -1,58 +1,85 @@
-import json
 import logging
 import re
 
-from streamlink.plugin import Plugin, PluginError
+from streamlink.plugin import Plugin
+from streamlink.plugin.api import validate
 from streamlink.stream import HLSStream
+from streamlink.utils import parse_json
 
 log = logging.getLogger(__name__)
 
 
 class Vlive(Plugin):
-    _url_re = re.compile(r"https?://(?:www.)vlive\.tv/video/(\d+)")
-    _video_status = re.compile(r'oVideoStatus = (.+)<', re.DOTALL)
-    _video_init_url = "https://www.vlive.tv/video/init/view"
+    _url_re = re.compile(r"https?://(?:www\.)vlive\.tv/(?P<format>video|post)/(?P<id>[0-9\-]+)")
```

Comment:
```suggestion
    _url_re = re.compile(r"https://www\.vlive\.tv/(?P<format>video|post)/(?P<id>[0-9\-]+)")
```
if there is only https support you don't need to ignore the `s`
same for the `www` subdomain